apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx

```diff
@@ -36,19 +36,21 @@ const UsernameTextfield = (props: ICustomUsernameProps & Partial<React.Component
   const [usernameIsAvailable, setUsernameIsAvailable] = useState(false);
   const [markAsError, setMarkAsError] = useState(false);
   const [openDialogSaveUsername, setOpenDialogSaveUsername] = useState(false);
-
+  const [finalUserNameValue, setFinalUserNameValue] = useState('');
   const debouncedApiCall = useMemo(
     () =>
       debounce(async (username) => {
+        username = slugify(username);
         const { data } = await fetchUsername(username);
         setMarkAsError(!data.available);
         setUsernameIsAvailable(data.available);
+        setFinalUserNameValue(username);
       }, 150),
     []
   );
 
   useEffect(() => {
-    if (!inputUsernameValue) {
+    if (!inputUsernameValue || !slugify(inputUsernameValue)) {
```

Comment:
Adding !slugify(inputUsernameValue) will handle cases like these:

User Input => '+++'
value of !inputUsernameValue = false
value of !slugify(inputUsernameValue) = true because this function would return '' for this user input.
It will enter the condition and cancel the debounce preventing unneccesary api call.